lib/WP_Auth0_Email_Verification.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * Class WP_Auth0_Email_Verification
+ */
+class WP_Auth0_Email_Verification {
+
+	private static $resend_nonce_action = 'auth0_resend_verification_email';
+
+	/**
+	 * Setup hooks tied to functions that can be dequeued
+	 */
+	public static function init() {
+		add_action( 'wp_ajax_nopriv_resend_verification_email', 'wp_auth0_ajax_resend_verification_email' );
+	}
+
+	/**
+	 * Stop the login process and show email verification prompt
+	 *
+	 * @param object $userinfo
+	 */
+	public static function render_die( $userinfo ) {
+
+		$html = sprintf(
+			'<p>%s<br><a id="js-a0-resend-verification" href="#">%s</a></p>
+			<p><a href="%s?%d">%s</a></p>
+			<script>var WPAuth0EmailVerification={ajaxUrl:"%s",sub:"%s",nonce:"%s",e_msg:"%s",s_msg:"%s"}</script>
+			<script src="%s"></script>
+			<script src="%s"></script>',
+
+			__( 'Please verify your email and log in again.', 'wp-auth0' ),
+			__( 'Resend verification email.', 'wp-auth0' ),
+			wp_login_url(),
+			time(),
+			__( '← Login', 'wp-auth0' ),
+			esc_url( admin_url( 'admin-ajax.php' ) ),
+			esc_js( $userinfo->sub ),
+			esc_js( wp_create_nonce( self::$resend_nonce_action ) ),
+			esc_js( __( 'Something went wrong; please login and try again.', 'wp-auth0' ) ),
+			esc_js( __( 'Email successfully re-sent to ' . $userinfo->email . '!', 'wp-auth0' ) ),
+			'//code.jquery.com/jquery-1.12.4.js',
+			WPA0_PLUGIN_URL . 'assets/js/die-with-verify-email.js?ver=' . WPA0_VERSION
+		);
+
+		$html = apply_filters( 'auth0_verify_email_page', $html, $userinfo, '' );
+
+		wp_die( $html );
+	}
+
+	/**
+	 * AJAX handler to request that the verification email be resent
+	 * Triggered in $this->render_die
+	 */
+	public static function ajax_resend_email() {
+
+		check_ajax_referer( self::$resend_nonce_action, 'nonce' );
+
+		$connect_info = WP_Auth0_Api_Client::get_connect_info();
+
+		$token = WP_Auth0_Api_Client::get_token(
```

Comment:
is it needed to request a new token on each resend trigger?

Reply:
@lbalmaceda - No, not required ... should probably be cached in a number of places.

Reply:
can you fix this on this same PR?